docs/guide.rst

```diff
@@ -31,109 +31,179 @@ The `WGPU <https://github.com/pygfx/wgpu-py>`_ library provides the low level AP
+    cube = gfx.Mesh(geometry, material)
+    scene.add(cube)
 
-Setting up a renderer
----------------------
+Objects are slightly more complicated than lights or cameras. They have a
+`geometry`, which controls an object's form, and a `material`, which controls an
+object's appearance (color, reflectiveness, etc). From here, we can hand our
+result to `gfx.show` and visualize it. This time, however, we are passing a `Scene`
+instead of an `Object`, so the result will look a little different::
 
-To render your scene to the canvas, you need a renderer. A renderer
-needs a target to render to (which can be a canvas or a texture).
-And finally, you need a camera to define the point of view to render the scene from.
+    gfx.show(scene)
 
-.. code-block:: python
+.. image:: _static/guide_static_cube.png
 
-    # A renderer is associated with a canvas (or a texture) that it renders to
-    renderer = gfx.renderers.WgpuRenderer(canvas)
+This happens because a `Scene` can be rendered as-is, whereas an `Object` can
+not. As such, `gfx.show` will, when given an `Object`, create a new scene for
+us, add the missing lights, a camera, and a background (for visual appeal),
+place the object into the scene and then render the result. When given a
+`Scene`, on the other hand, it can use the input as-is, allowing you to see
+exactly what you've created and potentially spot any problems.
 
-    # A camera defines the viewpoint and projection
-    camera = gfx.PerspectiveCamera(70, 16 / 9)
+**Canvases**
 
-    ...
+The second main building block is the `Canvas`. A `Canvas` provides the surface
+onto which the scene should be rendered and to use it you directly import it
+from wgpu-py (on top of which pygfx is built). Wgpu-py has several canvases that
+we can choose from, but for starters the most important ones is ``auto``, which
+automatically selects an appropriate backend to talk to the GPU and the OS::
 
-    # The actual rendering
-    renderer.render(scene, camera)
+    # @almarklein: Can we provide a more meaningful example
+    import pygfx as gfx
+    from wgpu.gui.auto import WgpuCanvas
 
+    canvas = WgpuCanvas()
+    cube = gfx.Mesh(
+        gfx.box_geometry(200, 200, 200),
+        gfx.MeshPhongMaterial(color="#336699"),
+    )
 
-Putting it together
--------------------
+    if __name__ == "__main__":
+        gfx.show(cube, canvas=canvas)
 
-If you run this, you should see a rotating yellow cube.
+.. image:: _static/guide_hello_world.png
 
-.. code-block:: python
+If you run the above, you will notice that it looks exactly the same as our
+introductory example. Just as with a `Scene` above, `gfx.show` will create a
+`Canvas` for you unless you provide it explicitly, and by default this uses the
+``auto`` backend and hence we won't see any changes.
 
-    import pygfx as gfx
+Why then would we explicitly provide a `Canvas`? Well, beyond the `Canvas`
+itself a backend also provides an event loop. This event loop allows us to
```

Comment:
I think by far the most common use-case for providing a custom canvas is when someone builds a GUI, and wants gfx to render into a subwidget. If think should just say that here. It speaks more to the imagination, and now the user knows that PyGfx can indeed do that. The part about the event loop is true, but may be confusing to newcomers.